Block/CategoryMapping.php

```diff
@@ -0,0 +1,137 @@
+use Magento\Store\Model\Store;
+use Magento\Store\Model\StoreManagerInterface;
+use Nosto\Tagging\Logger\Logger as NostoLogger;
+use Nosto\Tagging\Model\Service\Product\Category\DefaultCategoryService as CategoryBuilder;
+use Magento\Catalog\Model\Category;
+use Magento\Framework\Exception\LocalizedException;
+
+class CategoryMapping extends Template
+{
+    /** @var StoreManagerInterface */
+    private $storeManager;
+
+    /** @var CollectionFactory */
+    private $collectionFactory;
+
+    /** @var CategoryBuilder */
+    private $categoryBuilder;
+
+    /** @var NostoLogger */
+    private $logger;
+
+    /**
+     * CategoryMapping constructor.
+     */
+    public function __construct(
+        StoreManagerInterface $storeManager,
+        CollectionFactory $collectionFactory,
+        CategoryBuilder $categoryBuilder,
+        Context $context,
+        NostoLogger $logger
+    ) {
+        parent::__construct($context);
+        $this->storeManager = $storeManager;
+        $this->collectionFactory = $collectionFactory;
+        $this->categoryBuilder = $categoryBuilder;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @return false|string
+     */
+    public function getCategoryMap() {
+
+        $array = [];
+        try {
+            $store = $this->storeManager->getStore();
+            if ($store instanceof Store) {
+                $array = $this->getMagentoCategories($store);
+            }
+        } catch (NoSuchEntityException $e) {
+            $this->logger->exception('Could not get store');
+        }
+
+        return json_encode((object) $array, JSON_UNESCAPED_SLASHES);
+    }
+
+    /**
+     * @param Store $store
+     * @return array
+     */
+    private function getMagentoCategories(Store $store) {
+
+        $baseUrl = '';
+        $categoriesArray = [];
+
+        try {
+            $baseUrl = $store->getBaseUrl();
+
+            $categories = $this->collectionFactory->create()
+                ->addAttributeToSelect('*')
+                ->addAttributeToFilter('include_in_menu', array('eq' => 1))
+                ->addIsActiveFilter()
+                ->setStore($store);
+
+            /** @var Category $category $item */
+            foreach ($categories->getItems() as $category) {
+                $nostoCategoryString = strtolower(
+                    $this->categoryBuilder->getCategory($category, $store)
+                );
+                $categoryUrl = $baseUrl . '' . $category->getUrlPath();
+                if ($nostoCategoryString) {
+                    $categoriesArray[$nostoCategoryString] = $categoryUrl;
+                }
+            }
+        } catch (\Exception $e) {
+            $this->logger->exception(sprintf("Could not fetch base url for store %s",
+                $store->getName())
+            );
+        } catch (LocalizedException $e) {
+            $this->logger->exception('Could not filter category collection');
+        }
```

Comment:
The second catch will never happen, `\Exception` is the base class of every exception, should always be the last case in a try-catch.